### PR TITLE
MAINT: Move to new numpy.random API in test_*.py

### PR DESCRIPTION
**Pull request by MridulS**
This PR is another go at https://github.com/astropy/astropy/pull/12669 and this builds on top of https://github.com/astropy/astropy/pull/13747. With this PR we can start addressing https://github.com/astropy/astropy/issues/11687

This PR does not make any API changes! This just edits the tests file to update them to the old numpy random generator API.

There is a ruff numpy rule, so I am trying to make it happy here. For certain tests I had to play around with seeds, atol/rtol. I still need to work a bit more on this.

Once this is in we can start looking at adding arguments to expose numpy RNG instance for functions that depend on internal random state. Once this is in we can also start deprecating `NumpyRNGContext`.

### CODE REVIEW

**Comment by pllim** [comment]
Out of scope here but I just want to link for future self... https://github.com/astropy/astropy/issues/11566

File: astropy/units/tests/test_quantity_array_methods.py [hunk 1]
```diff
@@ -553,8 +553,8 @@ def test_slice_to_quantity(self):
         """
         Regression test for https://github.com/astropy/astropy/issues/2003
         """
-
```
Comment (nstarman) [inline astropy/units/tests/test_quantity_array_methods.py:556]: @MridulS, what if you defined a `rng` fixture in the main `conftest.py`, then you can just add an `rng` argument to any pytest function to get the rng. Should save a lot of lines!

**Review by nstarman — APPROVED** [review]
LGTM for `units`. However, see my other comment for a way to simplify all the tests using a `rng` pytest fixture.

**Comment by neutrinoceros** [comment]
https://github.com/astropy/astropy/issues/11687 Seems uncrontroversial so I assume the reason this was never merged is again that sweeping changes are hard to get approved. I'll open a tracker issue and will start re-issuing this patch in per-subpackage bits.

**Pull request closed without merging**